pennylane/pulse/transmon.py

```diff
@@ -215,3 +217,202 @@ def __add__(self, other):
+
+    where :math:`[a^\dagger_p, a_q] = i \delta_{pq}` are bosonic creation and annihilation operators
+    and :math:`q` the qubit label (``wires``).
+    The parameters ``amplitude``. ``phase`` and ``freq`` correspond to :math:`\Omega`, :math:`phi`
+    and :math:`nu`, respectively, and can all be either fixed numbers (``float``) or depend on time
+    (``callable``).
+
+    For more realistic simulations, one may restrict the amplitude to at most
+    :math:`\Omega_{\text{max}} = 20 \text{MHz}` and the carrier frequency to deviate at most
+    :math:`\nu - \omega = \pm 1 \text{GHz}` from the qubit frequency :math:`\omega`
+    (see :func:`~.transmon_interaction` and `arXiv:2203.06818 <https://arxiv.org/abs/2203.06818>`_).
+    The phase :math:`\phi(t)` is typically a slow function of time compared to :math:`\Omega(t)`.
+
+    .. note:: Currently only supporting ``d=2`` with qudit support planned in the future.
+
+    .. seealso::
+
+        :func:`~.drive`, :func:`~.rydberg_drive`, :func:`~.transmon_interaction`
+
+    Args:
+        omega (Union[float, list[float]]): List of dressed qubit frequencies in GHz. Needs to match the length of ``wires``.
+            When passing a single float all qubits are assumed to have that same frequency.
+        connections (list[tuple(int)]): List of connections ``(i, j)`` between qubits i and j.
+            When the wires in ``connections`` are not contained in ``wires``, a warning is raised.
+        g (Union[float, list[float]]): List of coupling strengths in GHz. Needs to match the length of ``connections``.
+            When passing a single float need explicit ``wires``.
+        anharmonicity (Union[float, list[float]]): List of anharmonicities in GHz. Ignored when ``d=2``.
+            When passing a single float all qubits are assumed to have that same anharmonicity.
+        wires (Union[int, list[int]]): Label of the qubit that the drive acts upon. Can be a list of multiple wires.
+        d (int): Local Hilbert space dimension. Defaults to ``d=2`` and is currently the only supported value.
+
+    Returns:
+        HardwareHamiltonian: a :class:`~.ParametrizedHamiltonian` representing the transmon interaction
+
+    **Example**
+
+    TODO: simple catch-all example
+
+    TODO: simple integration example with transmon_interaction
+
+    """
+    if d != 2:
+        raise NotImplementedError(
+            "Currently only supporting qubits. Qutrits and qudits are planned in the future."
+        )
+
+    wires = Wires(wires)
+
+    # TODO: use sigma+ and sigma- (not necessary as terms are the same, but for consistency)
+    # We compute the `coeffs` and `observables` of the EM field
+    coeffs = [
+        AmplitudeAndPhaseAndFreq(qml.math.cos, amplitude, phase, freq),
+        AmplitudeAndPhaseAndFreq(qml.math.sin, amplitude, phase, freq),
+    ]
+
+    drive_x_term = 0.5 * sum(qml.PauliX(wire) for wire in wires)
```

Comment:
Where does the `0.5` come from? :thinking: 
$$\exp[i(\phi(t)+\nu t)] a_q + \exp[-i(\phi(t)+\nu t)] a_q^\dagger$$
$$= \cos(\phi(t)+\nu t) (a_q+a_q^\dagger)+\sin(\phi(t)+\nu t) (ia_q-ia_q^\dagger) $$
$$= \cos(\phi(t)+\nu t) X - \sin(\phi(t)+\nu t) Y$$

Reply:
This might be due to some convention I am not aware of, maybe related to some rescaling to obtain the bosonic commutation relation? :thinking:

Reply:
The factor 0.5 comes from the bloqade docs https://queracomputing.github.io/Bloqade.jl/dev/hamiltonians/ indeed the factor should not be here, at least not if we keep the docs as is

Reply:
Ah, okay! Are we communicating this clearly? I believe documentation is the superpower against all trouble arising from unit/prefactor conventions :)

Reply:
It is clearly communicated in the sense that the docs provide precise definitions (i.e. providing how we treat $a=1/2(\sigma^x + i \sigma^y)$.

Now that you say it, it might be worth pointing out in the "theoretical background" section of `drive` that there is a factor 1/2 👍 https://docs.pennylane.ai/en/latest/code/api/pennylane.pulse.drive.html

Reply:
I also added a note in `transmon_drive` to be sure, you are right 👍